tla/trace2scen.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+
+import sys
+import json
+import os
+
+def comment(action):
+    return f"# {action['name']} {action['location']['module']}:{action['location']['beginLine']}"
+
+def term(ctx, pre):
+    return str(pre["currentTerm"][ctx['i']])
+
+def noop(ctx, pre, post):
+    return ["# Noop"]
+
+MAP = {
+    "ClientRequest": lambda ctx, pre, post: ["replicate", term(ctx, pre), "42"],
+    "MCClientRequest": lambda ctx, pre, post: ["replicate", term(ctx, pre), "42"],
+    "CheckQuorum": lambda ctx, pre, post: ["periodic_one", ctx['i'], "110"],
+    "Timeout": lambda ctx, pre, post: ["periodic_one",  ctx['i'], "110"],
+    "MCTimeout": lambda ctx, pre, post: ["periodic_one",  ctx['i'], "110"],
+    "RequestVote": noop,
+    "AppendEntries": lambda _, __, ___: ["dispatch_all"],
+    "BecomeLeader": noop,
+    "SignCommittableMessages": lambda ctx, pre, post: ["emit_signature", term(ctx, pre)],
+    "MCSignCommittableMessages": lambda ctx, pre, post: ["emit_signature", term(ctx, pre)],
+    "ChangeConfigurationInt": lambda ctx, pre, post: ["replicate_new_configuration", term(ctx, pre), *ctx["newConfiguration"]],
+    "AdvanceCommitIndex": noop,
```

Comment:
Can we make these print `# Noop (AdvanceCommitIndex)`, rather than identical `# Noop`s? Would mean we get a clearer correspondence to each trace action, without counting noops.

If they dominate real generated scenarios, I'd also be tempted to just omit lines for these entirely.

Reply:
We can, but they each immediately follow a comment that contains the action, the line etc already (see comment()), so it's a bit redundant. The rationale for comment + actions is that there can be more than one action emitted (like for bootstrap).

There are quite a few, but I think it's good to keep them for clarity.